mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -865,14 +865,14 @@ public Map<SchedulingConstraints, List<TaskExecutorAllocationRequest>> getGroupe
                 .collect(Collectors.groupingBy(TaskExecutorAllocationRequest::getConstraints));
         }
 
-        public Map<Double, Integer> getGroupedByCoresCount() {
+        public Map<SchedulingConstraints, Integer> getGroupedByConstraintsCount() {
             return allocationRequests
                 .stream()
-                .collect(Collectors.groupingBy(a -> a.getConstraints().getMachineDefinition()))
+                .collect(Collectors.groupingBy(TaskExecutorAllocationRequest::getConstraints))
                 .entrySet()
                 .stream()
                 .collect(Collectors.toMap(
-                    e -> e.getKey().getCpuCores(),
+                    Entry::getKey,
                     e -> e.getValue().size(),
                     Integer::sum
                 ));
```

Comment:
nit: isn't this just the previous method with the size of the array as the value? it feels like we are duplicating the logic.

Reply:
👍